e2e/_suites/kubernetes-autodiscover/autodiscover_test.go

```diff
@@ -0,0 +1,437 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/cucumber/godog"
+	messages "github.com/cucumber/messages-go/v10"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/elastic/e2e-testing/internal/shell"
+	"github.com/elastic/e2e-testing/internal/utils"
+)
+
+const defaultBeatVersion = "8.0.0-SNAPSHOT"
+const defaultEventsWaitTimeout = 120 * time.Second
+const defaultDeployWaitTimeout = 120 * time.Second
+
+type podsManager struct {
+	kubectl kubernetesControl
+	ctx     context.Context
+}
+
+func (m *podsManager) executeTemplateFor(podName string, writer io.Writer, options []string) error {
+	path := filepath.Join("testdata/templates", sanitizeName(podName)+".yml.tmpl")
+
+	usedOptions := make(map[string]bool)
+	funcs := template.FuncMap{
+		"option": func(o string) bool {
+			usedOptions[o] = true
+			for _, option := range options {
+				if o == option {
+					return true
+				}
+			}
+			return false
+		},
+		"beats_namespace": func() string {
+			return utils.GetDockerNamespaceEnvVar("beats")
+		},
+		"beats_version": func() string {
+			return shell.GetEnv("GITHUB_CHECK_SHA1", shell.GetEnv("BEAT_VERSION", defaultBeatVersion))
```

Comment:
I added this in https://github.com/elastic/e2e-testing/pull/1064/commits/dcd98735d70ec77a7194ac459c4b30bfde2a878c to support using the versions from beats PRs, would this be enough? I guess docker inside kind needs to be authenticated too? :thinking:

Reply:
@mdelapenya sorry, I wanted to ping you in this comment ^ :slightly_smiling_face: